### PR TITLE
Clean network code

### PR DESCRIPTION
**Pull request by CJKkkk-315**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #12390

## What this PR does / why we need it:
清理网络代码

###### 2. 改进读数据方法
- 优化内存分配次数
- 将读协议数据和处理逻辑解耦。
###### 3. 改进写数据方法
- 输出缓冲区改用定长buffer+变长buffer
- 减少写操作的次数。
- 修复了当字段大小等于16MB时会引发的Bug。
- Write方法改为只写状态包(OK,ERR,EOF)，执行时先Flush Buffer内数据。
- 改进Load Local
###### 4. 改进UT
- 适配新代码。修改UT及Proxy部分代码

https://github.com/matrixorigin/ci-test/actions/runs/9854565566

___

### **PR Type**
Bug fix, Enhancement, Tests, Other

___

### **Description**
- Decoupled handshake logic for better modularity.
- Improved data reading methods:
  - Optimized memory allocation.
  - Decoupled protocol data reading from processing logic.
- Enhanced data writing methods:
  - Switched to fixed-length and variable-length buffers.
  - Reduced write operations.
  - Fixed a bug triggered when field size equals 16MB.
  - Modified Write method to only write status packets (OK, ERR, EOF) and flush buffer data before execution.
  - Improved Load Local functionality.
- Refactored MySQL protocol implementation and tests to use custom `Conn` type instead of `goetty.IOSession`.
- Updated various tests to use `net.Pipe` for simulating network connections.
- Added comprehensive tests for MySQL protocol read and write operations.
- Commented out old tests and added placeholders for future implementations.
- Fixed a typo in the `recordRoutine` method name.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>12 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>mysql_protocol_test.go</strong><dd><code>Refactor MySQL protocol tests for improved connection handling</code></dd></summary>
<hr>

pkg/frontend/mysql_protocol_test.go

<li>Added new functions <code>registerConn</code>, <code>createInnerServer</code>, and <br><code>startInnerServer</code> for server setup and connection handling.<br> <li> Replaced <code>echoServer</code> with <code>startInnerServer</code> for server initialization in <br>tests.<br> <li> Updated tests to use <code>net.Pipe</code> for client-server connection simulation.<br> <li> Commented out or removed unused code and functions.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-e6b33be301283d85ec5e07d87fa0d6d826a68d7810019b613d315659f329ebac">+474/-438</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>query_result_test.go</strong><dd><code>Update query result tests for improved connection handling</code></dd></summary>
<hr>

pkg/frontend/query_result_test.go

<li>Updated test setup to use <code>net.Pipe</code> for client-server connection <br>simulation.<br> <li> Removed usage of <code>mock_frontend.NewMockIOSession</code> in favor of the new <br><code>Conn</code> type.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-c1e0b09d973ae5e5f7da681f1df18bc9f46424ec353b0cf03d8c7c5cab1b7530">+8/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mysql_buffer_test.go</strong><dd><code>Add tests for MySQL protocol read and write operations</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/mysql_buffer_test.go

<li>Added comprehensive tests for MySQL protocol read and write <br>operations.<br> <li> Included tests for small and large packets, as well as edge cases.<br> <li> Utilized <code>net.Pipe</code> for simulating network connections.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-9a35cec3edadfaefd26292c61fd51c20074b93cfdf01a30c8a7837f5dc303759">+640/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mysql_cmd_executor_test.go</strong><dd><code>Update MySQL command executor tests to use real connections</code></dd></summary>
<hr>

pkg/frontend/mysql_cmd_executor_test.go

<li>Replaced mock IOSession with real <code>net.Pipe</code> connections.<br> <li> Updated tests to use <code>NewIOSession</code> for session creation.<br> <li> Adjusted test cases to reflect changes in session handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-2f31f40e6021882a0487cd8dfbf5bbafc7b4f2c244b5e3a2ee27e62e36412107">+115/-88</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>session_test.go</strong><dd><code>Update session tests to use real connections</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/session_test.go

<li>Updated session tests to use <code>net.Pipe</code> for simulating network <br>connections.<br> <li> Replaced mock IOSession with <code>NewIOSession</code>.<br> <li> Adjusted test cases to reflect changes in session handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-b536d27dc9147be10e04f8209df0846ee094ada16301d8c0d6c3256569504c0a">+42/-34</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>authenticate_test.go</strong><dd><code>Update authentication tests to use real connections</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/authenticate_test.go

<li>Updated authentication tests to use <code>net.Pipe</code> for simulating network <br>connections.<br> <li> Replaced mock IOSession with <code>NewIOSession</code>.<br> <li> Adjusted test cases to reflect changes in session handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-1ad0724d270095c61cc27acd6e6178cdb16bd6fd041f8ccd081b0ebb82b470ee">+28/-27</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>protocol_test.go</strong><dd><code>Update protocol tests to use real connections</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/protocol_test.go

<li>Updated protocol tests to use <code>net.Pipe</code> for simulating network <br>connections.<br> <li> Replaced mock IOSession with <code>NewIOSession</code>.<br> <li> Adjusted test cases to reflect changes in session handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-a760a9a73edbf74bf4a790628b0143b652a3c81a18db3ca7e2edc7b131c79ebe">+29/-15</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>routine_test.go</strong><dd><code>Update routine tests to use real connections</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/routine_test.go

<li>Updated routine tests to use <code>net.Pipe</code> for simulating network <br>connections.<br> <li> Added connection registration for testing.<br> <li> Adjusted test cases to reflect changes in session handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-00b0717bbd26d9959134b0bfe56c9590814c3ba72937578b4dedcabd42dbc941">+29/-5</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>server_conn_test.go</strong><dd><code>Update test CN server to use NewIOSession</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/proxy/server_conn_test.go

<li>Updated test CN server to use <code>NewIOSession</code> for session creation.<br> <li> Adjusted test cases to reflect changes in session handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-985ebb23a7720ac11c6747252dbd4956cc453b070d1dc483f77db978b7ea6e88">+6/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>client_conn_test.go</strong><dd><code>Update client connection tests to use NewIOSession</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/proxy/client_conn_test.go

<li>Updated client connection tests to use <code>NewIOSession</code> for session <br>creation.<br> <li> Adjusted test cases to reflect changes in session handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-90e29f2b45f3e866ded12d1547f69f9b45156e076061976ae06c8e2ac08adb8c">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>query_tcp.result</strong><dd><code>Update expected TCP packet count in test result</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/distributed/cases/zz_statement_query_type/query_tcp.result

- Updated expected TCP packet count in test result.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-318a55c3f80bf1e71ab3683d1356bd5d74f17c85af7b5237d1327f09a825621a">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>query_tcp.sql</strong><dd><code>Update expected TCP packet count in test SQL</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/distributed/cases/zz_statement_query_type/query_tcp.sql

- Updated expected TCP packet count in test SQL.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-8059e86930015675e8f62dc82dd6168f889e8ac79e7bc94b908917b7d66c981e">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>12 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>mysql_protocol.go</strong><dd><code>Refactor MySQL protocol implementation for custom connection handling</code></dd></summary>
<hr>

pkg/frontend/mysql_protocol.go

<li>Replaced <code>goetty.IOSession</code> with custom <code>Conn</code> type for connection <br>handling.<br> <li> Refactored packet handling functions to return errors.<br> <li> Removed unused functions and variables.<br> <li> Updated functions to use the new <code>Conn</code> methods for reading, writing, <br>and managing packets.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-df9bbc34115212918609e47ad31cf72bf7b2be07d21c34c045074def608d0462">+410/-415</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>server.go</strong><dd><code>Refactor server implementation for custom connection handling</code></dd></summary>
<hr>

pkg/frontend/server.go

<li>Replaced <code>goetty.NetApplication</code> with custom listener and connection <br>handling.<br> <li> Added functions for starting and stopping listeners and handling <br>connections.<br> <li> Refactored handshake and message handling logic to use the new <code>Conn</code> <br>type.<br> <li> Updated server initialization to use <code>net.Listen</code> for TCP and Unix <br>socket connections.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-0c7b4946e699b8ac99537041944416f31a895bcf22fbc2660b79239a1a5fdcae">+275/-34</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mysql_buffer.go</strong><dd><code>Implement MySQL protocol buffer management</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/mysql_buffer.go

<li>Implemented a new <code>Conn</code> struct for managing MySQL protocol buffers.<br> <li> Added methods for reading, writing, and managing buffer memory.<br> <li> Introduced <code>NewIOSession</code> for creating new IO sessions.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-57df9142267c7468f8fc339e9071fe9750d24b5a6ab2efbca5d0a2765ff5ca59">+504/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>routine_manager.go</strong><dd><code>Replace goetty.IOSession with custom Conn type in routine manager</code></dd></summary>
<hr>

pkg/frontend/routine_manager.go

<li>Replaced <code>goetty.IOSession</code> with custom <code>Conn</code> type.<br> <li> Updated routine management methods to use the new <code>Conn</code> type.<br> <li> Fixed a typo in <code>recordRoutine</code> method name.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-511d6a4e67914d6a32b75c81d76cf23e8d6d255822eb34ccc298b84acfb52e8e">+12/-179</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mysql_cmd_executor.go</strong><dd><code>Refactor processLoadLocal to use new Read method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/mysql_cmd_executor.go

<li>Refactored <code>processLoadLocal</code> to use the new <code>Read</code> method.<br> <li> Simplified packet reading and writing logic.<br> <li> Improved error handling and logging.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-af2611d5fc89704398fe09d09644efa41fec8931b395eda292f2f474f1216275">+45/-63</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>internal_executor.go</strong><dd><code>Add placeholder methods for Read and Free in internalProtocol</code></dd></summary>
<hr>

pkg/frontend/internal_executor.go

<li>Added placeholder methods for <code>Read</code> and <code>Free</code> in <code>internalProtocol</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-692597348c3f3829194d5e28ec7f8e3d55ea1eccad94e9d29273bb04f56be337">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>protocol.go</strong><dd><code>Replace goetty.IOSession with custom Conn type in MysqlProtocolImpl</code></dd></summary>
<hr>

pkg/frontend/protocol.go

<li>Replaced <code>goetty.IOSession</code> with custom <code>Conn</code> type in <code>MysqlProtocolImpl</code>.<br> <li> Updated <code>Flush</code> method to use the new <code>Conn</code> type.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-b3b3d5650e84c5f1a197735136882995997f1b3e02392d550a3af22a556f7d0f">+3/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>types.go</strong><dd><code>Update MysqlReader interface with Read and Free methods</code>&nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/types.go

<li>Updated <code>MysqlReader</code> interface to include <code>Read</code> and <code>Free</code> methods.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-5426ed8bd66760011299bcef0a9d1d552477eaae49a8451e16b8ec5bf7d7310f">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>routine.go</strong><dd><code>Remove goetty.IOSession parameter from NewRoutine</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/routine.go

<li>Removed <code>goetty.IOSession</code> parameter from <code>NewRoutine</code>.<br> <li> Updated routine initialization to reflect changes in session handling.<br> <br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-b6993169af5bac764ab2a6ee1819633e1bc85f1561f01c22e31a4e586e34dc5c">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>client_conn.go</strong><dd><code>Update client connection to use NewIOSession</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/proxy/client_conn.go

<li>Updated client connection to use <code>NewIOSession</code> for session creation.<br> <li> Adjusted connection initialization to reflect changes in session <br>handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-c44eab21fd68effeed137d993e217bf5dd627f85dd4d6ea171343a9797041cf5">+6/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>server_conn.go</strong><dd><code>Update server connection to use NewIOSession</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/proxy/server_conn.go

<li>Updated server connection to use <code>NewIOSession</code> for session creation.<br> <li> Adjusted connection initialization to reflect changes in session <br>handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-92aff84b74bca918d9549cc14e086b41d1977aabd22836c7752a6802ae2a63aa">+6/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>handshake.go</strong><dd><code>Update TLS upgrade process to use UseConn method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/proxy/handshake.go

<li>Updated TLS upgrade process to use <code>UseConn</code> method for session <br>handling.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-9ce918f3edc0b12d791664199a6ba207361303b9fc6d4d2df23fd35432bdc577">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Miscellaneous</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>iopackage_test.go</strong><dd><code>Comment out old echo server and client test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/iopackage_test.go

<li>Commented out the old echo server and client test.<br> <li> Added placeholders for future implementation.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-cf5028352add8550184db85a516278b545fc4606d4ad1d7fac09516a3727bb27">+97/-108</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>routine_manager_test.go</strong><dd><code>Comment out old routine manager test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/routine_manager_test.go

<li>Commented out the old routine manager test.<br> <li> Added placeholders for future implementation.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-90e7e07d002fe74763c0d2df840f151d21d2d5c4d7fa6df911d2896b1478f3a5">+29/-35</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>codec.go</strong><dd><code>Remove unused imports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/codec.go

- Removed unused imports.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-8776e7caad32db9198d497007fb370ebab9ebf6274ee9b43b00023c27c80c70b">+1/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>cn.toml</strong><dd><code>Add newline at end of cn.toml</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

etc/launch/cn.toml

- Added a newline at the end of the file.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-d19a9d5ce28886197f25a7a772e39b836a8300b13adac462bc9db570bcea06a5">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>session.go</strong><dd><code>Fix typo in recordRoutine method name</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/frontend/session.go

- Fixed a typo in `recordRoutine` method name.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17459/files#diff-88f0aa40a4ccdc2de0ba0760f6e37fd2627e2ae337489de5ac58dcc9075bd7f8">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: etc/launch/cn.toml [hunk 1]
```diff
@@ -12,3 +12,4 @@ port-base = 18000
 check-fraction = 65536
 enable-metrics = true
 
+
```
Comment (daviszhen) [inline etc/launch/cn.toml:15]: 删掉这个修改。

File: pkg/frontend/iopackage_test.go [hunk 2]
```diff
@@ -235,102 +224,102 @@ func setServer(val int32) {
...
-
-	to := NewTimeout(1*time.Minute, false)
-	for isClosed() && !to.isTimeout() {
-	}
-	time.Sleep(15 * time.Second)
-	echoClient()
-	setServer(1)
-	wg.Wait()
-}
+//func echoHandler(session goetty.IOSession, msg interface{}, received uint64) error {
```
Comment (daviszhen) [inline pkg/frontend/iopackage_test.go:227]: 不需要了，删掉吧。

File: pkg/frontend/mysql_cmd_executor.go [hunk 3]
```diff
@@ -2120,61 +2119,24 @@ func processLoadLocal(ses FeSession, execCtx *ExecCtx, param *tree.ExternParam,
...
-		ses.Errorf(execCtx.reqCtx,
-			"Failed to load local file",
-			zap.String("path", param.Filepath),
-			zap.Error(err))
-	}
+	start := time.Now()
 	epoch, printEvery, minReadTime, maxReadTime, minWriteTime, maxWriteTime := uint64(0), uint64(1024*60), 24*time.Hour, time.Nanosecond, 24*time.Hour, time.Nanosecond
-	for {
+
+	readThenWrite := func() error {
```
Comment (daviszhen) [inline pkg/frontend/mysql_cmd_executor.go:2129]: 剥离成单个函数。

File: pkg/frontend/mysql_protocol.go [hunk 4]
```diff
@@ -624,6 +560,7 @@ func (mp *MysqlProtocolImpl) SetSession(ses *Session) {
 	mp.m.Lock()
 	defer mp.m.Unlock()
 	mp.ses = ses
+	mp.tcpConn.ses = ses
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:563]: ！！！在Close的时候，要主动mp.tcpConn.ses = nil。短链接会内存leak

File: pkg/frontend/protocol.go [hunk 5]
```diff
@@ -293,5 +290,5 @@ func (mp *MysqlProtocolImpl) EnableAutoFlush() {
 }
 
 func (mp *MysqlProtocolImpl) Flush() error {
-	return nil
+	return mp.tcpConn.Flush()
```
Comment (daviszhen) [inline pkg/frontend/protocol.go:293]: 这个flush和 上面的AutoFlush相关的接口和字段都删了。

**Review by daviszhen — CHANGES_REQUESTED** [review]

**Comment by daviszhen** [comment]
整体上在预期内。Thank you

File: pkg/sql/parsers/dialect/mysql/mysql_sql.go [hunk 6]
```diff
@@ -13478,7 +13478,7 @@ yydefault:
 //line mysql_sql.y:3478
 		{
 			assignments := []*tree.VarAssignmentExpr{
-				&tree.VarAssignmentExpr{
+				{
```
Comment (daviszhen) [inline pkg/sql/parsers/dialect/mysql/mysql_sql.go:13481]: 这个文件都是自动生成的。不需要修改呀。

Comment (daviszhen) [reply pkg/sql/parsers/dialect/mysql/mysql_sql.go:13481]: 在这个pr里面，应该涉及不到parser的修改！！！

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/frontend/routine_manager_test.go [hunk 7]
```diff
@@ -14,64 +14,50 @@
...
-	time.Sleep(100 * time.Millisecond)
-	cf.Close()
-
-	err = mo.Stop()
-	require.NoError(t, err)
-	wg.Wait()
-
-	closeDbConn(t, db)
-}
+//func create_test_server() *MOServer {
```
[the remark this answers is not in the document]
Comment (CJKkkk-315) [reply pkg/frontend/routine_manager_test.go:17]: 已恢复

File: pkg/frontend/mysql_cmd_executor.go (hunk 3 above)
Comment (CJKkkk-315) [reply pkg/frontend/mysql_cmd_executor.go:2129]: 不使用闭包的话 会有太多参数传递 感觉不太好

File: pkg/frontend/codec.go [hunk 8]
```diff
@@ -16,12 +16,10 @@ package frontend
 
 import (
 	"context"
-	"io"
-
 	"github.com/fagongzi/goetty/v2/buf"
 	"github.com/fagongzi/goetty/v2/codec"
-
 	"github.com/matrixorigin/matrixone/pkg/common/moerr"
+	"io"
```
Comment (daviszhen) [inline pkg/frontend/codec.go:22]: 这个文件的代码还有需要的吗？不要就把这个文件删了。

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/frontend/mysql_cmd_executor.go (hunk 3 above)
Comment (daviszhen) [reply pkg/frontend/mysql_cmd_executor.go:2129]: > 不使用闭包的话 会有太多参数传递 感觉不太好

改掉吧

File: pkg/frontend/codec.go (hunk 8 above)
Comment (CJKkkk-315) [reply pkg/frontend/codec.go:22]: proxy需要通过codec

File: pkg/frontend/mysql_cmd_executor.go (hunk 3 above)
Comment (CJKkkk-315) [reply pkg/frontend/mysql_cmd_executor.go]: > > 不使用闭包的话 会有太多参数传递 感觉不太好
> 
> 改掉吧

已修改

**Review by daviszhen — APPROVED** [review]

**Pull request closed without merging**